File: src/frontend/storefront/storefront/src/app/core/model/order-summary.ts
import { OrderSummaryItem } from './order-summary-item';
import { PaymentDetails } from './payment-details';

export interface OrderSummary {
  id: number;
  userId: string;
  status: string;
  total: number;
  itemCount: number;
  createdAt: string;
  items: OrderSummaryItem[];
  payment: PaymentDetails;
}

File: src/backend/platform/api/src/main/java/dev/agasen/api/core/order/read/OrderSummary.java
package dev.agasen.api.core.order.read;

import dev.agasen.api.core.payment.read.PaymentDetails;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

public record OrderSummary(
   Long id,
   String userId,
   String status,
   BigDecimal total,
   int itemCount,
   Instant createdAt,
   List< OrderSummaryItem > items,
   PaymentDetails payment
) {

   public OrderSummary( OrderDetails orderDetails, PaymentDetails paymentDetails, List< OrderSummaryItem > items ) {
      this(
         orderDetails.getId(),
         orderDetails.getUserId(),
         orderDetails.getStatus(),
         orderDetails.getTotal(),
         orderDetails.getItemCount(),
         orderDetails.getCreatedAt(),
         items,
         paymentDetails
      );
   }

}
